cmd/contour/serve.go

```diff
@@ -206,22 +200,17 @@ func doServe(log logrus.FieldLogger, ctx *serveContext) error {
 	// using the SyncList to keep track of what to sync later.
 	var informerSyncList k8s.InformerSyncList
 
-	informerSyncList.Add(dynamicInformerFactory.ForResource(ingressroutev1.IngressRouteGVR).Informer()).AddEventHandler(dynamicHandler)
-	informerSyncList.Add(dynamicInformerFactory.ForResource(ingressroutev1.TLSCertificateDelegationGVR).Informer()).AddEventHandler(dynamicHandler)
-	informerSyncList.Add(dynamicInformerFactory.ForResource(projectcontour.HTTPProxyGVR).Informer()).AddEventHandler(dynamicHandler)
-	informerSyncList.Add(dynamicInformerFactory.ForResource(projectcontour.TLSCertificateDelegationGVR).Informer()).AddEventHandler(dynamicHandler)
+	// Build the default GVR to Informer lookup object
+	gvri := k8s.NewGVRInformerLookupDefaults(dynamicInformerFactory, ctx.UseExperimentalServiceAPITypes)
+
+	// Use the GVR to Informer lookup map to register all the required informers.
+	for _, inf := range gvri.Lookup {
+		informerSyncList.Add(inf).AddEventHandler(dynamicHandler)
```

Comment:
maybe if informerSyncList grew a new method, `RegisterInformer` then this comment wouldn't be needed.

Reply:
Fair enough, changed `Add` to `RegisterInformer`.